### DISCUSSION TITLE
heic:speed value

### DISCUSSION THREAD

**Discussion by MimoKar** [ImageMagick/ImageMagick#5902] [Help] [answered]
### ImageMagick version

7,1,0.55

### Operating system, version and so on

linux

### Description

Hello,

Just to be sure: heic:speed=0 is placebo level or ultrafast?

**Comment by urban-warrior** [comment, marked as answer]
We pass the `speed` parameter directly to the HEIF delegate library.  Any behavior changes based on this parameter could best be answered by the `libheif` documentation or the `libheif` developers.

**Comment by tlacroix** [comment]
I had the same question, so I experimented a bit. To answer OP's question: It is not.

With my arbitrary reference PNG with alpha, the image took to convert to AVIF:

Without `heic:speed`, it took 9.69 secs — 4.275% size reduction
With `heic:speed=0`, it took 67.40 secs — 4.015% size reduction
With `heic:speed=2`, it took 35.65 secs — 3.970% size reduction
With `heic:speed=5`, it took 9.82 secs — 4.275% size reduction
With `heic:speed=8`, it took 3.48 secs — 4.538% size reduction

I ran the same test 5 times per `heic:speed` level (and 5 times without).

The default hence seems to be `heic:speed=5`.

The nature of my test image probably has a huge impact on the compression ratio, but the `(size reduction / compression time)` ratio seems to be proportional to the `heic:speed` parameter.

My 2 cents.

**Comment by MimoKar** [comment]
Hi, thanks for your test.
But in my case, this parameter doesn't show any difference now as the file size is same in both cases and even, there is no difference in speed.
I am converting to HEIC, not to AVIF.

**Reply by MimoKar** [reply]
AS per code in libheif, the preset parameter for x265 library is used as string not integer, and I don't see any conversion there, so this might be the reason that it is not affecting HEIC conversion.

**Reply by tlacroix** [reply]
From my understanding (which may very well be wrong), HEIC and HEIF and AVIF images have the same encoding, just different containers. Just like you can have an H.265 stream in an MP4, or an MKV, or an MOV file. And that, in any case, when it comes to converting a JPEG or PNG image to any of them (HEIC/HEIF/AVIF), it's the same encoder doing the same job, just the "packaging" (file format) being different. But I may very well be wrong.

Anyhow, thanks for pointing out the HEIC/HEIF/AVIF difference in any case. I honestly didn't catch you were working with HEIC and not AVIF: I was focused on the command line parameter (`heic:speed=N`).

I settled for `heic:speed=8` for my project; but I had to put it on the back-burner because the resulting files aren't consistenly properly decoded by most recent versions of major browsers supporting AVIF. I also tried `heif-enc` with similar performance, and/but different consistency issues. Looks like the encoding tools aren't as mature as for WebP just yet. Maybe soon.

**Reply by tlacroix** [reply]
Last note, because I very well know we are off topic (as per urban-warrior's initial reply), I was using the `convert` command-line tool on Linux and MacOS, not the Python or any other API.
